### DISCUSSION TITLE
Some expressions are available in Perspective-Python but not in the Monaco editor

### DISCUSSION THREAD

**Discussion by iscanegemen** [perspective-dev/perspective#1670] [Q&A] [answered]
While using the **expressions** feature, I have noticed that there are some functions which are available in Perspective-Python, but not in the built-in monaco editor from Perspective-Viewer. 

The list of functions that monaco editor does not recognize is the following: 

- _search, indexof, replace, replace_all, match, match_all, substring_

There are several questions that I need help with:

1. As long as I can understand, all of these functions fall in the category of string processing. Are there other functions from, let's say numeric operators, that I cannot use only in the monaco editor.
2. Is there a way for me to use these functions in the monaco editor, or are there other functions which would replace the usage of these functions? For example how can we integrate a substring into our expression without using the substring operator?

**Comment by texodus** [comment, marked as answer]
These seem to work for me on `1.0.6`.  I believe an earlier version may have neglected to add these to monaco's validator - can you check again on e.g. [blocks](https://bl.ocks.org/texodus/45b868833c9f456bd39a51e606412c5d)?

<img width="558" alt="Screen Shot 2021-12-11 at 4 19 13 PM" src="https://user-images.githubusercontent.com/60666/145693372-45a1f529-1cf5-4547-aa5e-20ec7bfe0396.png">

**Reply by iscanegemen** [reply]
Thanks for the fast reply. 

You are right, these operations work with the monaco editor from the link. I will try to update my version as soon as possible. Also, the monaco editor of [Perspective's website](https://perspective.finos.org/) may not be up-to-date as well.

**Reply by texodus** [reply]
You're right, the version on the website is out-of-date, we'll get the newest docs up this week.
